v8isolate.cc

```diff
@@ -20,6 +20,18 @@ V8Isolate::V8Isolate() {
   isolate_ = v8::Isolate::New(create_params);
 }
 
+V8Isolate::V8Isolate(const char* snapshot_data) {
+  startup_data = v8::V8::CreateSnapshotDataBlob(snapshot_data);
```

Comment:
I believe you are responsible for freeing the resulting memory.  I think you are currently leaking the snapshot blobs.